### ISSUE TITLE
Clean up use of unittest assert functions in twisted/cred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8379]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8379|
|Type|enhancement|
|Created|2016-06-03 02:03:25Z|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/cred
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8379 8379
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464919405215153 1464919405215153
changetime__1466479061877603 1466479061877603
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/95

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc removed owner|
|-|-|

OK, I uploaded a new diff which just replaces deprecated aliases:
https://github.com/twisted/twisted/pull/95

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez set owner to @rodrigc|
|-|-|

Reviewed, good to go.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset 551ac986831bb871dc2f2c068ce06d69955fcff1
```
#!CommitTicketReference repository="" revision="551ac986831bb871dc2f2c068ce06d69955fcff1"
Merge '8379-rodrigc-assert': Clean up use of unittest assert functions

Author: rodrigc
Reviewer: wsanchez
Fixes: #8379
```
